src/query_parser/v2/parser.y

```diff
@@ -865,11 +860,9 @@ query ::= text_expr(A) ARROW LSQB vector_query(B) RSQB . { // main parse, hybrid
 
 query ::= star ARROW LSQB vector_query(B) RSQB . { // main parse, simple vecsim search as entire query case.
   setup_trace(ctx);
-  switch (B->vn.vq->type) {
-    case VECSIM_QT_KNN:
-      B->vn.vq->knn.order = BY_SCORE;
-      break;
-  }
+  RS_LOG_ASSERT(B->vn.vq->type == VECSIM_QT_KNN, "vector_query must be KNN");
+  B->vn.vq->knn.order = BY_SCORE;
```

Comment:
add comment to explain why is different

Reply:
When the query is "pure" KNN query (with no other filters) - we can return the results from vecsim lib ordered by score and avoid sorting by id (this is an optimization)